src/main/java/com/eprosima/fastdds/idl/templates/DDSPubSubTypeSource.stg

```diff
@@ -76,11 +76,11 @@ struct_type(ctx, parent, struct) ::= <<
 $if (parent.IsInterface)$$parent.name$_$endif$$struct.name$PubSubType::$if (parent.IsInterface)$$parent.name$_$endif$$struct.name$PubSubType()
 {
     $if(ctx.GenerateTypesROS2)$setName("$struct.ROS2Scopedname$");$else$setName("$struct.scopedname$");$endif$
-    uint32_t type_size = $struct.cScopedname$_max_cdr_typesize;
+    size_t type_size = $struct.cScopedname$_max_cdr_typesize;
```

Comment:
I will keep `uint32_t`, because `TopicDataType::m_typeSize` is `uint32_t` and maybe there is loss of data.